packages/components/src/ui/color-picker/use-deprecated-props.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * External dependencies
+ */
+import colorize, { ColorFormats } from 'tinycolor2';
+// eslint-disable-next-line no-restricted-imports
+import type { ComponentProps } from 'react';
+import memoize from 'memize';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type ColorPicker from './component';
+
+type ColorPickerProps = ComponentProps< typeof ColorPicker >;
+
+/**
+ * @deprecated
+ */
+type LegacyColor =
+	| string
+	| {
+			hex: string;
+			hsl: ColorFormats.HSL | ColorFormats.HSLA;
+			hsv: ColorFormats.HSV | ColorFormats.HSVA;
+			rgb: ColorFormats.RGB | ColorFormats.RGBA;
+			/**
+			 * @deprecated
+			 */
+			oldHue: number;
+			/**
+			 * @deprecated
+			 */
+			source: 'hex';
+			draftHex: string;
+			draftHsl: ColorFormats.HSL | ColorFormats.HSLA;
+			draftRgb: ColorFormats.RGB | ColorFormats.RGBA;
+	  };
+
+/**
+ * @deprecated
+ */
+export interface LegacyProps {
+	color: LegacyColor;
+	/**
+	 * @deprecated
+	 */
+	onChangeComplete: ( colors: LegacyColor ) => void;
+	/**
+	 * @deprecated
+	 */
+	oldHue: string;
+	className: string;
+	/**
+	 * @deprecated
+	 */
+	disableAlpha: boolean;
+}
+
+function isLegacyProps( props: any ): props is LegacyProps {
+	return typeof props.onChangeComplete !== 'undefined';
+}
+
+function getColorFromLegacyProps(
+	props: LegacyProps
+): ColorFormats.HSL | ColorFormats.HSLA {
+	if ( typeof props.color === 'string' ) {
+		return colorize( props.color ).toHsl();
+	}
+
+	return props.color.hsl;
+}
+
+function toHsv(
+	color: colorize.Instance
+): ColorFormats.HSV | ColorFormats.HSVA {
+	const { h, s, v, a } = color.toHsv();
+
+	return {
+		h: Math.round( h ),
+		s: Math.round( s * 100 ),
+		v: Math.round( v * 100 ),
+		a,
+	};
+}
+
+const transformHslToLegacyColor = memoize(
+	( hsla: ColorFormats.HSL | ColorFormats.HSLA ): LegacyColor => {
+		const color = colorize( hsla );
+		const rawHex = color.toHex();
+		const rgb = color.toRgb();
+		const hsv = toHsv( color );
+		const hsl = hsla;
+
+		const isTransparent = rawHex === '000000' && rgb.a === 0;
```

Comment:
Why is `000000` considered transparent?

Reply:
No idea. This code is copied from the existing behavior. https://github.com/WordPress/gutenberg/blob/a7e2895829c16ecd77a5ba22d84f1dee1cfb0977/packages/components/src/color-picker/utils.js#L57

Reply:
Got it. I guess we should keep it as it was to preserve the "legacy" behaviour, even if this code doesn't really make sense to me.

Reply:
> Why is `000000` considered transparent?

Just had this trip me up in #34130. It's good to see that the color system is being updated :slightly_smiling_face: